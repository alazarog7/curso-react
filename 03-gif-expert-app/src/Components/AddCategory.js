import React, { useState } from 'react'
import PropType from 'prop-types';
const AddCategory = ({setcategory})=> {
    
    const [inputValue, setinputValue] = useState("");

    const handleInputChange = (e)=>{
       setinputValue(e.target.value);
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        if(inputValue.length > 0){
            setcategory([inputValue])
            setinputValue('');
        }
        
    } 

    return (
        <form onSubmit={handleSubmit}>
           <h1>{inputValue}</h1>
           <input 
                type="text" 
                value={inputValue}
                onChange = {handleInputChange} 
           />
        </form>
    )
}

AddCategory.propTypes = {
    setcategory: PropType.func.isRequired 
}


export default AddCategory;