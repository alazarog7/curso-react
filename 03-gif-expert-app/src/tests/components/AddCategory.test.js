import shallow from "enzyme/build/shallow"
import AddCategory from "../../Components/AddCategory"
import '@testing-library/jest-dom';

describe('AddCategory test', () => {
    const setcategory = jest.fn();
    let wrapper = shallow(<AddCategory setcategory={setcategory}/>);
    

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setcategory={setcategory}/>)
    })

    test('should return the reght template', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should return the right input value', () => {
        const input = wrapper.find('input');
        const valor = "ekeko";
        input.simulate('change',{target:{value:valor}});
        expect(wrapper.find('h1').text()).toBe(valor);
    })

    test('should not pot data on submit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setcategory).not.toHaveBeenCalled();
    })

    test('should call setcategories and clean input tex', () => {
        const input = wrapper.find('input');
        const valor = "ekeko";
        input.simulate('change',{target:{value:valor}});
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setcategory).toHaveBeenCalled();
        expect(setcategory).toHaveBeenCalledTimes(1);
        expect(setcategory).toHaveBeenCalledWith(expect.any(Array));

        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
    
    


    

})
