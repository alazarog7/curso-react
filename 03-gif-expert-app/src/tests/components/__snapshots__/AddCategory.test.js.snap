// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory test should return the reght template 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
